src/client.js

```diff
@@ -217,72 +227,60 @@ assign(Client.prototype, {
       helpers.warn('The pool has already been initialized')
       return
     }
-    this.pool = new Pool(assign(this.poolDefaults(config.pool || {}), config.pool))
+
+    const poolSettings = this.getPoolSettings(config.pool);
+
+    this.pool = genericPool.createPool(poolSettings.factory, poolSettings.config)
   },
 
   validateConnection(connection) {
-    return true
+    return Promise.resolve(true);
   },
 
   // Acquire a connection from the pool.
   acquireConnection() {
-    return new Promise((resolver, rejecter) => {
-      if (!this.pool) {
-        return rejecter(new Error('Unable to acquire a connection'))
-      }
-      let wasRejected = false
-      const t = setTimeout(() => {
-        wasRejected = true
-        rejecter(new Promise.TimeoutError(
+    if (!this.pool) {
+      return Promise.reject(new Error('Unable to acquire a connection'))
+    }
+    return this.pool.acquire()
+      .tap(connection => {
+        debug('acquired connection from pool: %s', connection.__knexUid)
+      })
+      .catch(genericPoolErrors.TimeoutError, () => {
+        throw new Promise.TimeoutError(
           'Knex: Timeout acquiring a connection. The pool is probably full. ' +
           'Are you missing a .transacting(trx) call?'
-        ))
-      }, this.config.acquireConnectionTimeout || 60000)
-      this.pool.acquire((err, connection) => {
-        clearTimeout(t)
-        if (err) {
-          return rejecter(err)
-        }
-        if (wasRejected) {
-          this.pool.release(connection)
-        } else {
-          debug('acquired connection from pool: %s', connection.__knexUid)
-          resolver(connection)
-        }
-      })
-    })
+        )
+      });
```

Comment:
Above only TimeoutError is catched, should also generic errors be catched and is there need to release connection from pool in some cases? (I'm just comparing, what code did earlier and what it seems to be doing now)

Reply:
Previously, `wasRejected` would only be `true` if timer `t` fired, in which case, `pool.acquire` may have resolved with a resource that needed to be released. Now that we simply call `pool.acquire`, there is no resource to be released in the `catch` handler.